swift/apple/FirezoneNetworkExtension/Adapter.swift

```diff
@@ -521,6 +523,7 @@ extension Network.NWPath {
     // We define a path as different from another if the following properties change
     return path.supportsIPv4 != self.supportsIPv4 || path.supportsIPv6 != self.supportsIPv6
       || path.supportsDNS != self.supportsDNS
+      || path.status != self.status
```

Comment:
Will this reset the moment we go offline or only when we come back online? Because on all other platforms, we reset when we go offline already.

Reply:
Based on testing, it depends - it's possible the OS may move us to a different "connected" state before moving to offline, in which case it'll reset before we go offline (path.status = unsatisfied).

But in other cases it may happen more immediately, i.e. transition directly to path.status = unsatisfied.

The changes I made ensure that we reset in both cases, so there is a chance we reset twice - once going offline, and again coming online.